lib/net/Makefile

```diff
@@ -5,20 +5,33 @@ NETIFFILES := $(filter-out $(LWIPDIR)/netif/slipif.c,$(NETIFFILES))
 LWIPSRCS := $(COREFILES) \
             $(CORE4FILES) \
             $(CORE6FILES) \
-            $(NETIFFILES) \
-            $(APIFILES)
+            $(APIFILES) \
+            $(NETIFFILES)
 
 # Include driver sources
 DRIVERSRCS := $(NXDK_DIR)/lib/net/pktdrv/pktdrv.c \
               $(NXDK_DIR)/lib/net/nforceif/src/driver.c \
               $(NXDK_DIR)/lib/net/nforceif/src/sys_arch.c
 
-SRCS += $(LWIPSRCS) $(DRIVERSRCS)
+NET_SRCS := \
```

Comment:
*(This is a generic comment about PR description, but made on code so we can resolve it more easily)*

> - Why can we do `#include <hal/debug.h>` but do `#include "lwip/debug.h"` in the samples? ("" vs <>) I switched it to <> and it still works. Note: I know the general difference between these two include methods, I just don't understand it in this context(nxdk, complex build systems, libs, etc. etc.).

I believe it often comes down to different conventions.

A library that expects to be bundled with the app will often use `""` and a library which is meant for system-wide install (or toolchain installation) is typically `<>`.
I assume lwip, as an embedded lib, is often bundled with the project (like a firmware), so `""` might make more sense then.

Since pkg-config and CMake came along those lines got blurred, because some people might have SDL systemwide, and others don't. Also some will have their SDL header in a path with other stuff, so now you have to listen to pkg-config or CMake (which recommend `#include "SDL.h"`, even if some platforms claim that it should be `#include <SDL2/SDL.h>`)

Different tools also handle it differently; here are the GCC rules: https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html
Some interesting notes:
- MSVC++ also likes to look in parent directories or something weird; so it finds files in different order. I believe you can even trip it by putting stuff in some root directory like `C:/`
- Some compilers ignore warnings when you use `<>`, others will not make this decision at time of use, but at time of include dir addition (like GCC with `-isystem` vs. `-i`).

I'm sure that @thrimbor will correct me on all of this though. He's usually more standard aware than I am.

---

Random rant of the day: With generic names like `debug.h`, `input.h`, `audio.h` etc. there can also be conflicts, which is why I absolutely hate this (which allows `#include "debug.h"` or `#include <debug.h>` when the only right thing should probably be `#include <hal/debug.h>`):
https://github.com/XboxDev/nxdk/blob/9519fbd8c259792e0fc56244c69b1f13a86fe97e/Makefile#L57

Reply:
This has caught me out a few times that 
I started changing it. 
My code had its own "debug.h" but was including the nxdk's one instead causing a confusimg set of compilation errors.

Maybe ill rebase and raise a PR.
https://github.com/Ryzee119/nxdk/pull/1/commits

Reply:
> Maybe ill rebase and raise a PR.
> https://github.com/Ryzee119/nxdk/pull/1/commits

Please upstream. I also had https://github.com/JayFoxRox/nxdk/pull/61 for a long time (also see failed https://github.com/XboxDev/nxdk/pull/258 for another approach).
I keep running into this and it gets even worse with autoconf and cmake (which is how I primarily use nxdk).

Reply:
> I'm sure that @thrimbor will correct me on all of this though. He's usually more standard aware than I am.

What you said sounds about right. The most notable difference afaik is that `<>` only searches the include directories given to the compiler while `""` searches relative to the code file first.

For nxdk, I think it's best to use `<>` everywhere when including a library's header, except for special cases, e.g. when including a header within a library that is meant to be private to the library or something.

> Random rant of the day: With generic names like `debug.h`, `input.h`, `audio.h` etc. there can also be conflicts, which is why I absolutely hate this (which allows `#include "debug.h"` or `#include <debug.h>` when the only right thing should probably be `#include <hal/debug.h>`)

Agreed. I recall adding that include directory because we moved the header and it conflicted with the way a submodule (SDL I believe) included it, but I don't remember whether we fixed that issue in the meantime. We should check and remove that include path if we can.

Reply:
Resolving after #461 has been merged